### PR TITLE
feat: filter templates by organization

### PR DESCRIPTION
**Pull request by aslilac**
Add support for filtering templates based on the organization they belong to.

As a bonus, since the backend already supports it, you can filter by `deprecated` and `name` as well.

Screenshot 2024-08-13 at 11 30 57 AM

### CODE REVIEW

File: site/src/pages/WorkspacesPage/WorkspacesPageView.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export interface WorkspacesPageViewProps {
   canChangeVersions: boolean;
 }
 
-export const WorkspacesPageView = ({
+export const WorkspacesPageView: FC<WorkspacesPageViewProps> = ({
```
Comment (jaaydenh) [inline site/src/pages/WorkspacesPage/WorkspacesPageView.tsx:68]: Any reason you explicitly want to use FC here? or is change to make this more consistent with other components?

**Comment by jaaydenh** [comment]
Nit: The alignment of the Templates header text and create template button changes if the org has templates or not. Not a high priority but just feels a bit odd when switching orgs.

Screenshot 2024-08-13 at 7 59 17 PM
Screenshot 2024-08-13 at 7 59 29 PM

**Comment by jaaydenh** [comment]
I am guessing there is no docs page yet for template filtering? Maybe, we shouldn't link to a docs page "View Advanced Filtering" until one exists?

Screenshot 2024-08-13 at 8 06 16 PM

**Review by BrunoQuaresma — APPROVED** [review]
The code looks good, but I'm missing tests for this feature.

Minor things that I found during QA:

1. Should we display the count information at the top of the table, as we do for workspaces and other pages with filters?

Screenshot 2024-08-14 at 10 01 49

2. I think we should only display the following content when it is empty, with no filters applied, or when there are no active templates.

Screenshot 2024-08-14 at 10 02 56

In my opinion, these are not blockers—perhaps the tests could be considered—but I'll trust your judgment, so I'm approving this. 👍

File: coderd/database/queries/templates.sql [hunk 2]
```diff
@@ -22,17 +22,23 @@ WHERE
 			organization_id = @organization_id
 		ELSE true
 	END
+	-- Filter by organization_name
+	AND CASE
+		WHEN @organization_name :: text != '' THEN
+			LOWER("organization_name") = LOWER(@organization_name)
+		ELSE true
+	END
```
Comment (Emyrk) [inline coderd/database/queries/templates.sql:30]: This needs to be removed.

The param on the frontend is `organization` and it accepts both an ID or an org name.

https://github.com/coder/coder/blob/604d5cb552449abbd9b650069a0e3bd6df9292b5/coderd/searchquery/search.go#L213-L227

File: site/src/pages/WorkspacesPage/WorkspacesPageView.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/pages/WorkspacesPage/WorkspacesPageView.tsx:68]: just consistency

**Comment by aslilac** [comment]
@BrunoQuaresma 

1. I'm going with no for now. This view isn't paginated, whereas the views that show this are. If we ever add pagination tho, definitely!

3. Yeah, we should probably tweak this, but this feels good enough for the experiment. I'll make a note about revisiting it for GA.

**Pull request merged**